editor.planx.uk/src/pages/FlowEditor/lib/store/navigation.ts

```diff
@@ -115,19 +118,28 @@ export const navigationStore: StateCreator<
   },
 
   /**
-   * Calculate the status of each section node based on the current position in a flow
-   *   ** will need to become aware of all breadcrumbs within a section in the near future for "back" rather than only currentCard & upcomingCardIds
+   * Calculate the status of each section node based on the current position in a flow (eg state.cachedBreadcrumbs) or provided breadcrumbs (eg on reconciliation)
+   *   Pass updatedSectionNodeIds on reconiliation to ensure we set an accurrate status for every previously seen section because updated sections are removed from breadcrumbs during reconciliation
    */
-  sectionStatuses: (): Record<string, SectionStatus> => {
+  sectionStatuses: (
+    breadcrumbs?: Store.breadcrumbs,
+    updatedSectionNodeIds?: string[]
```

Comment:
i had been going back on forth on this one but i think you're right that `updatedNodeIds` is much clearer here - and yes, because it's checking against SectionNodes anyways it's self-filtering in that sense. 

and it simplifies the Reconciliation Page logic a bit to just directly pass `Object.keys(data.removedBreadcrumbs)`